smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -1,451 +0,0 @@
-        }
-
-        fileManifest = context.getFileManifest();
-        LOGGER.info(() -> String.format("Generating TypeScript %s for service %s",
-                settings.generateClient() ? "client" : "server", service.getId()));
-
-        // Decorate the symbol provider using integrations.
-        SymbolProvider resolvedProvider = artifactType.createSymbolProvider(model, settings);
-        for (TypeScriptIntegration integration : integrations) {
-            resolvedProvider = integration.decorateSymbolProvider(model, settings, resolvedProvider);
-        }
-        symbolProvider = SymbolProvider.cache(resolvedProvider);
-
-        // Resolve the nullable protocol generator and application protocol.
-        protocolGenerator = resolveProtocolGenerator(integrations, service, settings);
-        applicationProtocol = protocolGenerator == null
-                ? ApplicationProtocol.createDefaultHttpApplicationProtocol()
-                : protocolGenerator.getApplicationProtocol();
-
-        writers = new TypeScriptDelegator(settings, model, fileManifest, symbolProvider, integrations);
-    }
-
-    private ProtocolGenerator resolveProtocolGenerator(
-            Collection<TypeScriptIntegration> integrations,
-            ServiceShape service,
-            TypeScriptSettings settings
-    ) {
-        // Collect all of the supported protocol generators.
-        Map<ShapeId, ProtocolGenerator> generators = new HashMap<>();
-        for (TypeScriptIntegration integration : integrations) {
-            for (ProtocolGenerator generator : integration.getProtocolGenerators()) {
-                generators.put(generator.getProtocol(), generator);
-            }
-        }
-
-        ShapeId protocolName;
-        try {
-            protocolName = settings.resolveServiceProtocol(model, service, generators.keySet());
-        } catch (UnresolvableProtocolException e) {
-            LOGGER.warning("Unable to find a protocol generator for " + service.getId() + ": " + e.getMessage());
-            protocolName = null;
-        }
-
-        return protocolName != null ? generators.get(protocolName) : null;
-    }
-
-    void execute() {
-        // Write shared / static content.
-        STATIC_FILE_COPIES.forEach((from, to) -> {
-            LOGGER.fine(() -> "Writing contents of `" + from + "` to `" + to + "`");
-            fileManifest.writeFile(from, getClass(), to);
-        });
-
-        // Generate models that are connected to the service being generated.
-        LOGGER.fine("Walking shapes from " + service.getId() + " to find shapes to generate");
-        Collection<Shape> shapes = new Walker(model).walkShapes(service);
-        for (Shape shape : TopologicalIndex.of(model).getOrderedShapes()) {
-            if (shapes.contains(shape)) {
-                shape.accept(this);
-            }
-        }
-        for (Shape shape : TopologicalIndex.of(model).getRecursiveShapes()) {
```

Comment:
> We want models_N.ts file only has import statement to its previous files.

Hmm, I didn't understand this mattered.

Shape generation already [does do topo sort](https://github.com/awslabs/smithy/blob/d8c3fb13403ae9ea7c0c8c7e4fffb8f64108f641/smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/directed/CodegenDirector.java#L377). 

This reordering is because [CodegenDirector does generation for Service before visiting shapes](https://github.com/awslabs/smithy/blob/d8c3fb13403ae9ea7c0c8c7e4fffb8f64108f641/smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/directed/CodegenDirector.java#L282-L284). I thought this rearrangement was not a big deal other than 1 time diff. But now I'm going to evaluate to confirm if it is good to switch the order of those steps in CodegenDirector for all generators - seems reasonable to me.

Reply:
Changing the order in Smithy - https://github.com/awslabs/smithy/pull/1289. Will regenerate SDKs once approved to confirm diffs are gone.